### PR TITLE
Verticalbar chart: Add Callout and Legends for vertical bar chart

### PR DESCRIPTION
**Pull request by RajeshGoriga**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes
Add Callout and Legends for vertical bar chart

#### Focus areas to test
Vertical bar chart on focus and on hover 

#### Screenshots
image

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/fluentui/pull/12623)

### CODE REVIEW

File: packages/charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx [hunk 1]
```diff
@@ -4,49 +4,198 @@ import { axisLeft as d3AxisLeft, axisBottom as d3AxisBottom, Axis as D3Axis } fr
...
+import { FocusZone, FocusZoneDirection } from '@fluentui/react-focus';
+import { ILegend, Legends } from '../Legends/index';
 
 import {
   IVerticalBarChartProps,
   IVerticalBarChartStyleProps,
   IVerticalBarChartStyles,
-  IDataPoint,
+  IVerticalBarChartDataPoint,
+  // IDataPoint,
```
Comment (Raghurk) [inline packages/charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx:17]: > // IDataPoint, [](start = 1, length = 15)

remove

File: packages/charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx [hunk 2]
```diff
@@ -4,49 +4,198 @@ import { axisLeft as d3AxisLeft, axisBottom as d3AxisBottom, Axis as D3Axis } fr
...
+      selectedLegendTitle: '',
+      refSelected: null,
+      dataForHoverCard: 0,
+      color: '',
+    };
+    this._refArray = [];
     this._adjustProps();
+  }
 
+  public render(): JSX.Element {
```
Comment (Raghurk) [inline packages/charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx:65]: >JSX.Element [](start = 19, length = 11)

can we change this to react.reactnode?

File: packages/charting/src/components/VerticalBarChart/__snapshots__/VerticalBarChart.test.tsx.snap [hunk 3]
```diff
@@ -13,55 +13,322 @@ exports[`VerticalBarChart renders VerticalBarChart correctly 1`] = `
...
+  >
+    <div
       className=
 
           {
-            transform: translate(30px, 0px);
+            align-items: center;
+            white-space: nowrap;
+            width: 100%;
           }
```
Comment (Raghurk) [inline packages/charting/src/components/VerticalBarChart/__snapshots__/VerticalBarChart.test.tsx.snap:113]: nit: Please format

File: packages/charting/src/components/VerticalBarChart/__snapshots__/VerticalBarChart.test.tsx.snap [hunk 4]
```diff
@@ -13,55 +13,322 @@ exports[`VerticalBarChart renders VerticalBarChart correctly 1`] = `
...
+                        }
+                  />
+                  <div
+                    className=
+
+                        {
+                          -moz-osx-font-smoothing: grayscale;
+                          -webkit-font-smoothing: antialiased;
+                          color: #000000;
+                          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
```
Comment (Raghurk) [inline packages/charting/src/components/VerticalBarChart/__snapshots__/VerticalBarChart.test.tsx.snap:229]: > font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif; [](start = 25, length = 144)

can we import this from fabric fonts so we have automatic fallback

File: packages/charting/src/components/VerticalBarChart/__snapshots__/VerticalBarChart.test.tsx.snap [hunk 5]
```diff
@@ -13,55 +13,322 @@ exports[`VerticalBarChart renders VerticalBarChart correctly 1`] = `
...
+                          opacity: ;
+                        }
+                  />
+                  <div
+                    className=
+
+                        {
+                          -moz-osx-font-smoothing: grayscale;
+                          -webkit-font-smoothing: antialiased;
+                          color: #000000;
```
Comment (Raghurk) [inline packages/charting/src/components/VerticalBarChart/__snapshots__/VerticalBarChart.test.tsx.snap:228]: > color: #000000; [](start = 25, length = 16)

theme colors from fabric

File: packages/charting/src/components/VerticalBarChart/__snapshots__/VerticalBarChart.test.tsx.snap [hunk 6]
```diff
@@ -13,55 +13,322 @@ exports[`VerticalBarChart renders VerticalBarChart correctly 1`] = `
...
+                      }
+                      &::-moz-focus-inner {
+                        border: 0;
+                      }
+                      .ms-Fabric--isFocusVisible &:focus:after {
+                        border: 1px solid transparent;
+                        bottom: 1px;
+                        content: "";
+                        left: 1px;
+                        outline: 1px solid #605e5c;
```
Comment (Raghurk) [inline packages/charting/src/components/VerticalBarChart/__snapshots__/VerticalBarChart.test.tsx.snap:279]: >     outline: 1px solid #605e5c; [](start = 19, length = 32)

fabric theme color

File: packages/charting/src/components/VerticalBarChart/__snapshots__/VerticalBarChart.test.tsx.snap (hunk 3 above)
Comment (RajeshGoriga) [reply packages/charting/src/components/VerticalBarChart/__snapshots__/VerticalBarChart.test.tsx.snap:105]: Formated

---
In reply to: [406300260](https://github.com/microsoft/fluentui/pull/12623#discussion_r406300260) [](ancestors = 406300260)

Comment (RajeshGoriga) [reply packages/charting/src/components/VerticalBarChart/__snapshots__/VerticalBarChart.test.tsx.snap:113]: This file is auto generated if any change done manual build is getting failed

**Pull request merged**